lib/parse.js

```diff
@@ -134,6 +135,7 @@ module.exports = function (str, opts) {
     options.depth = typeof options.depth === 'number' ? options.depth : defaults.depth;
     options.arrayLimit = typeof options.arrayLimit === 'number' ? options.arrayLimit : defaults.arrayLimit;
     options.parseArrays = options.parseArrays !== false;
+    options.decoder = typeof options.decoder === 'function' ? options.decoder : defaults.decoder;
```

Comment:
it might be a good idea to throw an explicit exception here if `options.decoder` is provided (not undefined), and not a function.